### DISCUSSION TITLE
How long will the Windows Server 2019 runner be available?

### DISCUSSION THREAD

**Discussion by DavidXanatos** [actions/virtual-environments#7840] [General questions] [answered]
How long will the Windows Server 2019 runner be available?
The Windows Server 2016 was deprecated some time ago and since there is Windows Server 2022 I wonder how long will GitHub offer the 2019 version?
I'm asking because as far as I see there is no easy way to compile a Windows 7 compatible driver using Visual Studio 2022 so if the 2019  runner gets axed one will have to install some 2019 build tools on the 2022 runner "manually" in order to build a windows 7 compatible driver, which will make the CI build process substantially slower.

**Comment by mikhailkoliada** [comment, marked as answer]
Hey!

@DavidXanatos we support two lates OS versions, so 2019 will be available as long as Windows server 2025 (I am guessing its version number) is not generally available

**Comment by LostRuins** [comment]
What do we do now...
